### ISSUE TITLE
Cannot get game controls to respond on ubisoft connect game

### ISSUE THREAD

**Issue by Autoxguy** [moraroy/NonSteamLaunchers-On-Steam-Deck#535]
I had installed ubisoft connect on my deck with this program back in February but had not actually logged into it until now. I then installed Pop the lost crown and it was able to launch but the controls were not responding. I am not sure what control scheme temate I need to use but I will try some things. 

Anyway, I tried rerunning the non steam launchers exe to reinstall the ubisoft connect app and I guess it used the most recent installer files to do so and once it finished it automatically took me to game mode.  When trying to run ubisoft connect from there it would start and then act like it would crash.  I then went back into desktop mode and noticed that I was not able to open the nonsteamgamelaunchers prefix folder like I was before (after I installed the game originally I mean) so that I could view the games files and such.  It's like the permissions got messed up or something. 

I then deleted  the nonsteamgame prefix folder, which may not have been a good idea.  I then reinstalled the ubi connect app via this program exe but now what is interesting is that if I go to install the game the space that is available is how much was left after installing the game the first time.  So for some reason it isn't seeing it even though it is there (apparently).

So in other words it's like the game files are still there but they cannot be seen or something.  Maybe I just to shut the deck down and restart it and maybe the correct amount of available space will be shown.

**Comment by moraroy** [comment]
@Autoxguy are the files "hidden"? you may need to enable hidden folders in the dolphin menu to see the compatdata folders . Open "Dolphin" and then in the top right hit the options button and go all the way down to view i think there you can select view hidden folders the full path would be something like "/home/deck/.local/steam/steamapps/compatdata/NonSteamLaunchers"

**Comment by Autoxguy** [comment]
Thank you for getting back to me.
Yes, I do have hidden folders enabled.  I did end up getting it to work but ended up getting the controls to work by NOT disabling the Connect overlay.  With not turning some things off in Connect the controls are able to work just fine.
The other thing I am not sure about, going back to the game data is, while Ubisoft Connect and the Lost crown game itself are installed in the NonSteamLaunchers folder under compatdata, there is another prefix folder above the NonSteamLaunchers folder that seems to have been created at the same time that has the naming convention of the 8 or 9 digits (the random longer number convention for the prefix that is given when you install a non-native steam game on the Deck) that also has a copy of Ubisoft Connect and a copy of the game as well.  So that is I think why now, I actually have two copies of the game on my deck, but only one of them is actually.
Is this normal, or can I delete the copy of the game (around 30GB) from the long numbered prefix folder (or maybe just delete that folder all together) and leave the NonSteamLaunchers folder and have everything be fine?

    On Thursday, December 19, 2024 at 12:26:17 AM EST, Roy ***@***.***> wrote:  
 
 


@Autoxguy are the files "hidden"? you may need to enable hidden folders in the dolphin menu to see the compatdata folders . Open "Dolphin" and then in the top right hit the options button and go all the way down to view i think there you can select view hidden folders the full path would be something like "/home/deck/.local/steam/steamapps/compatdata/NonSteamLaunchers"

**Comment by Autoxguy** [comment]
> @Autoxguy are the files "hidden"? you may need to enable hidden folders in the dolphin menu to see the compatdata folders . Open "Dolphin" and then in the top right hit the options button and go all the way down to view i think there you can select view hidden folders the full path would be something like "/home/deck/.local/steam/steamapps/compatdata/NonSteamLaunchers"

The bottom line is, yes I have hidden folders enabled.  The question comes down to, why is ubisoft connect and the game installed under the nonsteamlaunchers named prefix (which is what I expect to see based on how your script works), but at the same time ubisoft connect and the game are also installed under a random 9 digit prefix (like what non native steam games are installed under) as well. So ubisoft connect and the game are actually installed twice.  What is also weird is that if I delete the random number prefix then ubisoft connect won't boot up, the same thing happens when trying to delete the nonsteam launchers named prefix folder.
